File: frontend/api/comment.ts
import { GENERAL_IP, SSR_GENERAL_IP } from "@/constants/env";
import api from "./utils";

export const addComment = async (body: {
  articleId: number;
  title: string;
  content: string;
}) => {
  return await api.post(`${GENERAL_IP}/comment`, body);
};

export const getComments = async (
  userId: number,
  articleId: number,
  isSSR = false
) => {
  return await api.get(
    `${
      isSSR ? SSR_GENERAL_IP : GENERAL_IP
    }/comments?userId=${userId}&articleId=${articleId}`
  );
};

export const deleteComment = async (commentId: number) => {
  return await api.delete(`${GENERAL_IP}/comment?commentId=${commentId}`, {});
};

File: backend/general/entities/comment.go
package entities

import "time"

type Comment struct {
	CommentId  uint64    `gorm:"primaryKey"`
	UserId     uint64    `gorm:"column:userId"`
	ArticleId  uint64    `gorm:"column:article_id"`
	Title      string    `gorm:"column:title"`
	Content    string    `gorm:"column:content"`
	Edited     bool      `gorm:"column:edited"`
	UpdateTime time.Time `gorm:"column:update_time"`
}
